sdk/src/client/api/block_builder/transaction_builder/requirement/context_inputs.rs

```diff
@@ -1,26 +1,28 @@
 // Copyright 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use super::{Error, InputSelection};
+use super::{TransactionBuilder, TransactionBuilderError};
 use crate::{
     client::secret::types::InputSigningData,
     types::block::{
-        context_input::{BlockIssuanceCreditContextInput, CommitmentContextInput, RewardContextInput},
+        context_input::{BlockIssuanceCreditContextInput, CommitmentContextInput},
         output::{AccountId, DelegationOutputBuilder, Output},
     },
 };
 
-impl InputSelection {
-    pub(crate) fn fulfill_context_inputs_requirement(&mut self) -> Result<Vec<InputSigningData>, Error> {
+impl TransactionBuilder {
+    pub(crate) fn fulfill_context_inputs_requirement(
+        &mut self,
+    ) -> Result<Vec<InputSigningData>, TransactionBuilderError> {
         let mut needs_commitment_context = false;
 
-        for (idx, input) in self.selected_inputs.iter().enumerate() {
+        for input in &self.selected_inputs {
             match &input.output {
                 // Transitioning an issuer account requires a BlockIssuanceCreditContextInput.
                 Output::Account(account) => {
                     if account.features().block_issuer().is_some() {
                         log::debug!("Adding block issuance context input for transitioned account output");
-                        self.context_inputs.insert(
+                        self.bic_context_inputs.insert(
                             BlockIssuanceCreditContextInput::from(account.account_id_non_null(input.output_id()))
                                 .into(),
```

Comment:
I guess this and other .into() are not needed anymore

Reply:
tru